Wizard NPCs and UIV Event Rework

ErhardSteinhauer opened the pull request:
## About the PR
Overall, this update expands the loot pool for Wizard Federation NPCs by adding a variety of magical items with different levels of usefulness, new usable staves with unique projectile effects, and single-use spell scrolls. It also includes resprites for existing wizard gear. Some tables with snippets of information on current state of item sprites and UIV rework: https://docs.google.com/spreadsheets/d/1Z26YuT2nZrMT4Du6SxWfYLzCBK2LZebSCH2nn8vkNcw/edit?usp=sharing.

### New items
- **Items** (for the most part available as random loot from wizard gear bundles, but some have fixed spawns on Wizard UIV):
  - The Teapot of Infinite Tea: refills itself over time.
  - The Wallet of Boundless Greed: a wallet with a built-in magnet.
  - The Shaker: an endlessly shaking cocktail shaker. Also screams.
  - Animated Broom: a wizard-themed cleanbot.
  - ~~Sandals with Planewalker Socks: magboots that don’t require a grid (floor) to walk.~~ Canned for later PR.
  - “Bottomless” Gas Canister: a high-capacity gas canister (x10 of a normal one).
  - “Bottomless” Gas Tank: holds the equivalent of 1.5 canisters’ worth of gas.
  - Singing rock: wizard jukebox.
  - Potted Coin Tree: a tree that spawns coins.
  - ~~Jar of Angry Fairies: spawn small swarm of hostile mobs with timed despawn.~~ Canned for later PR.
  - Wizard's Portal Cloak of Holding: cloak with inventory.
  - Wizard's Sash of Holding: belt with bigger inventory.
  - Fairy Dragon Wings: wizard themed jetpack with big tank.
  - Ravenbar: a crowbar that can pry open bolted doors.
  - Wizard Scarf of Warmth: 95% protection from fire.
  - Infernal Core: wizard themed RTG.
  - Flying Broom: wizard themed hoverbike.
  - Wizard's Luggage Whistle: summons wizard's luggage - a crate that follows its conjurer, holds up to 40 items, slow.
  - Wizard's smoking pipes (two variants): one with regenerating THC other with regenerating nicotine. Normal corn cob pipe can be obtained from ShadyCigsDelux vendomat.
  - Metronome neck tie: makes you waddle if worn., no toggle, only waddle.
  - Wizard Stoles and Masks: plus new sprites for blue, red, and purple wizard robes (sprites by @GentleButter). Lootable directly from NPCs.
  - Staves. Lootable directly from NPCs.
  - Scrolls.

- **Staves**:
  - Inert staves dropped by wizard NPCs can now be activated by clicking on them with the correctly colored crystal shard.
    - For example, to activate an inert adept-wizard ice staff, use a cyan crystal shard to obtain an adept-wizard ice staff.
    - A staff can be further enhanced by applying an inert (or active) anomaly core to an already activated staff.
      - For example, to enhance an adept-wizard ice staff, use an anomaly core to obtain a master-wizard ice staff.
    - Master-wizard staves have more charges and can push NPCs in melee combat.
  - Added new rechargeable staves for wizards:
    - Ice Staff, Caustic Beam Staff, Fire Bolt Staff, Magic Bolt Staff, Blue Curaçao Elemental Staff, Lightning Staff, Soap Staff, and Brick Staff.
  - Added new rechargeable staves for Blood Cultists:
    - Crimson Bolt Staff, Malicious Touch Staff, Crimson Lightning Staff, and Shredding Soap Staff.

- **Scrolls**:
  - Added single-use scrolls that grant players temporary powers usable once. To gain the power, the scroll must first be unrolled and then read. Once unrolled, the scroll will disappear after 60 seconds.
    - Scrolls can be obtained as randomized loot from wizards and certain cultists.
    - Summoned NPCs are friendly to all players.
  - Added new scrolls:: 
    - Conjure blue curacao elemental scroll, fireball scroll, smoke scroll, force wall scroll (4 charges), conjure magivend scroll, flash rune scroll, ignite rune scroll, stun rune scroll, explosion rune scroll, repulse scroll, blink scroll, void applause scroll, knock scroll, recall scroll, ethereal jaunt scroll, conjure drained one scroll, conjure shade scroll.

- **New structures**:
  - Wizard staff altar: recharges activated wizard staves. Can be disassembled.
  - Blood cult staff altar: recharges activated blood cultist staves. Can be disassembled.
  - Conjured MagiVend: a temporary vending machine that despawns after 3 minutes; its inventory contains new items.
  - Floor decor: runes.
  - Wall-mounted crystal lighting fixtures.
  - MagiVend with Frontier specific wizard gear.

- **Changes to wizard NPCs**:
  - Split wizard NPCs into three tiers, similar to other hostile factions (no boss NPCs for now).
  - Enabled looting of outer clothing, masks, and scarves from wizard NPCs.
  - Reduced NPC health by 20%.
  - Added a damage modifier set with vulnerability to holy damage, applied to both Wizards and Blood Cultists. (Weapons with holy damage don’t exist yet, but they’ll be added later).

- **Changes to Wizard-themed UIV event**:
  - Reworked the hull from ground up. Features more walkable surface and indestructible walls and windows.
  - Fixed loot spawns: "bottomless" oxygen canister, "bottomless" nitrogen canister, "bottomless" gas tank (empty), ravenbar, wizard luggage whistle, tea pots of infinite tea, infinite candles, MagiVend, Fairy Dragon Wings, Singing Rock, Infernal Core, Coin Tree, Wallet of Boundless Greed.

## To Do
- [x] "Magic" items.
- [x] More staves.
- [x] Single use spell scrolls.
- [x] More scrolls.
- [x] More robes.
- [x] Cult specific altar for cultists staves.
- [x] Update Wizard-themed UIV.
- [x] Rework wizard NPCs loadouts and attacks.
- [x] Distribute new items among NPCs, update their loot bags.
- [x] Replace placeholder sprites on magical items.

## Why / Balance
The wizard-themed UIV is often avoided by players because the danger-to-reward ratio is heavily skewed toward danger. This PR aims to make the wizard-themed PvE content more rewarding and lays the groundwork for developing wizard NPCs into fully fledged dungeon and VGroid enemies. Some of the changes made to wizards have also been applied to Blood Cult NPCs, as they share several similar traits.

## Technical details
Lots of YML, some RSI, couple empty components.

## How to test
1. Spawn Wizard UIV (`addgamerule BluespaceWizardFederationScout`), check the NPCs, their loot bags, their weapons.
2. Spawn inert staves and various color crystals, grab crystals and click on staves. Some of staves will transform into ranged weapons. Use the "activated" staves.
3. Spawn staff recharging altars. Insert discharged staff into the altar.
4. Spawn single use spell scrolls, use them.

## Media
How to activate staves:
<img width="1287" height="302" alt="image" src="https://github.com/user-attachments/assets/2b023acd-b595-451f-b4ee-5b99667f07c6" />

<img width="834" height="836" alt="image" src="https://github.com/user-attachments/assets/82ddc107-20c5-461e-951b-cd97a75a1257" />

<img width="839" height="831" alt="image" src="https://github.com/user-attachments/assets/32d7a42b-fac3-4874-a23a-271dcb0e874a" />

<img width="466" height="462" alt="image" src="https://github.com/user-attachments/assets/a0a87769-2b3c-4c58-86d1-b670d5c68913" />

<img width="1344" height="960" alt="2025-10-19_01 03 29" src="https://github.com/user-attachments/assets/4d140740-f82a-4224-9c18-bf274974fc19" />
<img width="1344" height="960" alt="2025-10-19_01 03 24" src="https://github.com/user-attachments/assets/09f415c8-b697-415f-a939-36cb5cef6f88" />
<img width="1344" height="960" alt="2025-10-19_01 03 43" src="https://github.com/user-attachments/assets/f0f4f765-ba11-41a1-91a8-68076b91fc48" />
<img width="1344" height="960" alt="2025-10-19_01 03 49" src="https://github.com/user-attachments/assets/66b6cd0b-8900-4006-9088-c8ebb58ee74a" />
<img width="1344" height="960" alt="2025-10-19_01 03 54" src="https://github.com/user-attachments/assets/488b0818-1e68-4289-bfe7-0e6ececaa8c3" />
<img width="1344" height="960" alt="2025-10-19_01 04 00" src="https://github.com/user-attachments/assets/8c75f6e2-c43a-4540-a80a-1b3dbbe9ce6b" />


## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [x] I have read [CONTRIBUTING.md](https://github.com/new-frontiers-14/frontier-station-14/blob/master/CONTRIBUTING.md) and and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.
- [x] I have reviewed the [Ship Submission Guidelines](https://frontierstation.wiki.gg/wiki/Ship_Submission_Guidelines) if relevant.
- [x] I confirm that AI tools were not used in generating the material in this PR.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them. -->

**Changelog**
:cl:
- tweak: Reworked wizard NPCs and the wizard-themed UIV bluespace event.
- tweak: Wizards and Cultists are now vulnerable to Holy Damage.
- add: Added a variety of new magical items, spell scrolls, and staves.
- add: Added the Wizard Staff Altar and Blood Cult Staff Altar. New staves, after activation, can now be recharged at these altars.

JojyAsmir commented:
Extremely important last minute idea. Bluespace Pipe filled with effectively infinite nicotine and/or THC. No self respecting wizard should be without his pipe weed.

blackknight954 reviewed (changes requested):
I have given it a run through and the new map and items are amazing

but uhm
<img width="401" height="136" alt="image" src="https://github.com/user-attachments/assets/3f918038-773d-414a-b375-1021042dbe96" />
These either don't seem to produce enough power, produce power for the wrong type of cable, or the wiring is wrong somewhere, because the ship ran out of power and never recovered from it.

ErhardSteinhauer commented:
I probably should list the most recent changes:
- Switched the infernal core (generator) to HV cables.
- Replaced normal thrusters on UIV with rusted thrusters (no power draw).
- Mapped air alarms on UIV and linked them to atmos devices.
- Split LV cable network on UIV.
- Mapped vacuum markers on UIV.
- Mapped 2 flying brooms on UIV.
- Adjusted visuals on wizard wallet.

blackknight954 reviewed (approved):
Cool, all’s well then.

MagosAmphrose commented:
I love everything about this and can't wait for this to be merged. Particularly love the Luggage reference.

Looking forwards to seeing it ingame!

ErhardSteinhauer commented:
> probably need to add something to the spraypainter to prevent their use on the bottomless canisters

Or.. Or I can make them paintable:
<img width="200" height="302" alt="image" src="https://github.com/user-attachments/assets/dd8c83c7-230a-4bc3-8305-fbc8417d6edd" />

dustylens reviewed (approved):
All outstanding reported items appear to have been addressed.

Or disregarded entirely. Rude.

However everything that jumped out at us the last time we took a look seems to be resolved or ignored as described.
